### ISSUE TITLE
Finalising Spartacus parameter choices when coupled to SUEWS and deciding which should be user facing.

### ISSUE THREAD

**Issue by LewisB7** [UMEP-dev/SUEWS#49]
Ting and I discussed that before documenting Spartacus in the SUEWS manual, we should discuss parameter values. Also, we need to decide which parameters should be available to the SUEWS user.

I will document the parameters and highlight those which require discussion.

**Comment by LewisB7** [comment]
Please see attached table
[Spartacus_Parameters.docx](https://github.com/UMEP-dev/SUEWS/files/6453865/Spartacus_Parameters.docx)
.

**Comment by LewisB7** [comment]
[Spartacus_Parameters.docx](https://github.com/UMEP-dev/SUEWS/files/6478621/Spartacus_Parameters.docx)

I have updated the table in Spartacus_Parameters.docx and also included the namelist files. @suegrimmond @sunt05

**Comment by LewisB7** [comment]
I have made some small changes to Spartacus_Parameters.docx.

Following advice from Ting I have now put it in sharepoint:
https://livereadingac-my.sharepoint.com/:w:/g/personal/wa916280_reading_ac_uk/EbtFYHk0CytAgSJZQteBmo4BNcj6gi2uXgtPla7LLVz_aA?e=EEmn6C
so that we can all comment and edit.

**Comment by sunt05** [comment]
I decided to copy and put your linked document as md text here, which starts from the next post.

**Comment by sunt05** [comment]
Discussion points are highlighted in red. 
`RC` – RunControl.nml 
`SU` - Spartacus.nml 
`SUP` – Spartacus_Profiles.nml 
`I` - internal (i.e. not a user choice) 

## Implementation: 
`RunControl.nml` contains a parameter `NetRadiationMethod` and when set to larger than 1000 Spartacus will run. 
If when divided by 10 the remainder is 1, 2 or 3 then the Ldown method will be option 1, 2 and 3 in SUEWS respectively (i.e. the one used for `NetRadiationMethod = 1,2,3` respectively as described in https://suews.readthedocs.io/en/latest/input_files/RunControl/scheme_options.html#cmdoption-arg-NetRadiationMethod ) 
When `NetRadiationMethod>1000` Spartacus reads in `Spartacus.nml` and `Spartacus_Profiles.nml`. 
An if loop could be added within SUEWS so that different `NetRadiationMethod` values result in different Spartacus.nml and Spartacus_Profiles.nml files being brought in. 
These could have different parameter configurations. 

### `Spartacus.nml` contains constants and logicals: 
```
&Spartacus 
nlayers = 15 
use_sw_direct_albedo = false 
n_vegetation_region_urban = 1 
nsw = 1 
nlw = 1 
nspec = 1 
n_stream_sw_urban = 8 
n_stream_lw_urban = 8 
sw_dn_direct_frac = 0.0 
air_ext_sw = 0.0 
air_ssa_sw = 0.95 
veg_ssa_sw = 0.13 
air_ext_lw = 0.0 
air_ssa_lw = 0.0 
veg_ssa_lw = 0.01 
/ 
```

### `Spartacus_Profiles.nml` contains arrays of vertical profiles: 

```
&Spartacus_Profiles 
height(1,:) = 0.,  5., 10., 15., 20., 25., 30., 35., 40., 45., 50., 55., 60., 65., 70., 75 
building_frac(1,:) = 0.44079454, 0.43641054, 0.41948414, 0.3620058 , 0.27059461, 
        0.19314201, 0.1143779 , 0.07303782, 0.03994853, 0.02287435, 
        0.01801009, 0.01599049, 0.01091688, 0.00939603, 0.00845396 
veg_frac(1,:) = 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0. 
building_scale(1,:) = 34.95857866, 34.99503185, 35.1689881 , 36.6541438 , 40.42309674, 
       44.61352058, 44.69219304, 45.73676575, 43.63399501, 40.50083884, 
       40.4682398 , 41.07594181, 40.21164448, 42.41011588, 43.37860781 
veg_scale(1,:) = 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0,  
        10.0, 10.0, 10.0, 10.0, 10.0 
veg_ext(1,:) = 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 
        0.25, 0.25, 0.25, 0.25, 0.25 
veg_fsd(1,:) = 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5 
veg_contact_fraction(1,:) = 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0. 
roof_albedo(1,:) = 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 
        0.5, 0.5, 0.5, 0.5, 0.5 
wall_albedo(1,:) = 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 
        0.5, 0.5, 0.5, 0.5, 0.5 
ground_albedo_dir_mult_fact(1,:) = 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1. 
roof_albedo_dir_mult_fact(1,:) = 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1. 
wall_specular_frac(1,:) = 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0. 
roof_emissivity(1,:,1) = 0.91, 0.91, 0.91, 0.91, 0.91, 0.91, 0.91, 0.91, 0.91, 0.91, 
        0.91, 0.91, 0.91, 0.91, 0.91 
wall_emissivity(1,:,1) = 0.91, 0.91, 0.91, 0.91, 0.91, 0.91, 0.91, 0.91, 0.91, 0.91, 
        0.91, 0.91, 0.91, 0.91, 0.91 
/ 
```

**Comment by sunt05** [comment]
## Spartacus Parameters


Location&nbsp;in SUEWS | Parameter | Definition | Current&nbsp;Implementation | Action needed
-- | -- | -- | -- | --
RC | NetRadiationMethod | SpartacusUrban | 1003. For&nbsp;NetRadiationMethod&nbsp;&nbsp;&gt; 1000, when divided by 10, if the remainder equals 1, 2 or 3 then&nbsp;Ldown&nbsp;option is 1, 2 and 3, respectively.&nbsp;Kup&nbsp;and&nbsp;Lup&nbsp;are&nbsp;from&nbsp;Spartacus. | Update&nbsp;RunControl.nml
I | do_sw | Compute shortwave fluxes? | True | -
I | do_lw | Compute&nbsp;longwave fluxes? | True | -
SU | use_sw_direct_albedo | Specify ground and roof albedos separately for&nbsp;direct solar radiation? | False | Update&nbsp;Spartacus.nml
I | do_vegetation | Will vegetation be represented? | True&nbsp;if&nbsp;vegetation fraction &gt; 0 and False otherwise | -
I | do_urban | Will urban areas be represented? | True | -
SU | n_vegetation_region_urban | Number of regions used to describe vegetation&nbsp;(2 needed for heterogeneity) | 1 | Update&nbsp;Spartacus.nml
SU | nsw | Number of shortwave spectral intervals | 1 | Update&nbsp;Spartacus.nml
SU | nlw | Number of longwave spectral intervals | 1 | Update&nbsp;Spartacus.nml
SU | n_stream_sw_urban | SW diffuse streams per hemisphere | 2 | Update&nbsp;Spartacus.nml
SU | n_stream_lw_urban | LW streams per hemisphere | 2 | Update&nbsp;Spartacus.nml
I | ncol | Number of columns or tiles | 1 | -
SU | nlayers | Number of layers in column | Should be the same as&nbsp;the number of layers in&nbsp;the profile arrays (apart from height which is nlayers+1) | Update&nbsp;Spartacus.nml
I | i_representation | Surface type. 0: flat, 1: tree, 2: urban, 3:&nbsp;urban+tree | Calculate&nbsp;which surface type is present based on building and vegetation fractions | -
SU&nbsp;(maybe&nbsp;I&nbsp;since&nbsp;lw_emission&nbsp;calculations are not&nbsp;implemented&nbsp;for multiple bands&nbsp;in Spartacus yet) | nspec | Number of shortwave or longwave spectral&nbsp;intervals | 1&nbsp;(should be the same as&nbsp;nsw&nbsp;and&nbsp;nlw) | Update&nbsp;Spartacus.nml
SUP | height | Height of the layer interfaces | Nlay+1&nbsp;evenly separated&nbsp;layers&nbsp;between 0 and&nbsp;max building height. | Update&nbsp;Spartacus_Profiles.nml&nbsp;&nbsp;&nbsp;*** Fig to show difference&nbsp;between&nbsp;wind and&nbsp;temperature&nbsp;layers&nbsp; --&nbsp;which goes to RSL ***
I | ground_temperature | (K) | SUEWS&nbsp;TSfc_C. | -
I | roof_temperature | (K) | SUEWS&nbsp;TSfc_C. It is an&nbsp;Nlay&nbsp;array so could vary if T(z) becomes available in SUEWS. | -
I | wall_temperature | (K) | SUEWS&nbsp;TSfc_C. It is an&nbsp;Nlay&nbsp;array so could vary if T(z) becomes available in SUEWS. | -
I | clear_air_temperature | (K) | SUEWS&nbsp;temp_C. It is an&nbsp;Nlay&nbsp;array so could vary if T(z) becomes available in SUEWS. | -
I | veg_temperature | (K) | SUEWS&nbsp;temp_C. It is an&nbsp;Nlay&nbsp;array so could vary if T(z) becomes available in SUEWS. | -
I | veg_air_temperature | Temperature in the vegetation air space (K) | SUEWS&nbsp;temp_C. It is an&nbsp;Nlay&nbsp;array so could vary if T(z) becomes available in SUEWS. | -
SUP | building_fraction | Building plan area density | SUEWS&nbsp;sfr(2) | Update&nbsp;Spartacus_Profiles.nml&nbsp;&nbsp;&nbsp;Check&nbsp;Fr_Bldgs&nbsp;in SUEWS_SiteSelect.txt is the same as&nbsp;building_fraction&nbsp;first array index.
SUP | veg_fraction | Vegetation&nbsp;plan area density | SUEWS&nbsp;sfr(3)+sfr(4) | Update&nbsp;Spartacus_Profiles.nml&nbsp;&nbsp;&nbsp;Check&nbsp;Fr_EveTr&nbsp;+&nbsp;Fr_DecTr&nbsp;in SUEWS_SiteSelect.txt is the same as&nbsp;veg_fraction&nbsp;first array index.
SUP | building_scale | Building horizontal scale (m). Effective building diameter used (along with&nbsp;building_fraction) to calculate the average building perimeter length. Eq. 2 of spartacus_surface_documentation.pdf.&nbsp;&nbsp;Should be constant with height unless buildings get narrower with height. | Estimate from inspecting buildings in Google Maps or GIS?&nbsp;&nbsp;The exact method of estimation is more involved (and unrealistic that the general user would do it). The method is discussed in “An Exponential Model of Urban Geometry for Use in Radiative Transfer Applications”. | Update&nbsp;Spartacus_Profiles.nml
SUP | veg_scale | Vegetation horizontal scale (m)&nbsp;Vegetation scale used (along with&nbsp;veg_fraction) to calculate the average vegetation perimeter length. Eq. 1 of&nbsp;Spartacus_surface_documentation.pdf. | Estimate from inspecting buildings in Google Maps or GIS? | Update&nbsp;Spartacus_Profiles.nml
SUP | veg_ext | Wavelength independent&nbsp;vegetation&nbsp;extinction coefficient&nbsp;(m-1)&nbsp;(i.e.&nbsp;the number of radiance e-foldings&nbsp;per metre).&nbsp;&nbsp;Should this be related to LAI? At least one parameter in Spartacus should be.&nbsp;Hogan et al. 2018: “If the leaves can be assumed randomly oriented then the optical depth of the layer is equal to half the leaf area index (LAI)” | 0.25&nbsp;(just took from offline Spartacus Russell_square.nc) | Update&nbsp;Spartacus_Profiles.nml
SUP | veg_fsd | Fractional standard deviation of vegetation extinction | 0.5&nbsp;(just took from offline Spartacus&nbsp;config.nam) | Update&nbsp;Spartacus_Profiles.nml
SUP | veg_contact_fraction | Fraction of vegetation edge in contact with building walls | 0.0&nbsp;(choosing anything other than zero seems arbitrary without knowledge of the canopy geometry) | Update&nbsp;Spartacus_Profiles.nml
I | top_flux_dn_lw | Top-of-canopy downwelling longwave flux | Calculation of&nbsp;Ldown&nbsp;is set by the value of&nbsp;NetRadiationMethod&nbsp;in&nbsp;runcontrol.nml. | Update&nbsp;runcontrol.nml
SU | top_flux_dn_direct_sw | Top-of-canopy downwelling direct shortwave flux | 0.45*&nbsp;Top_flux_dn_lw&nbsp;as an estimate based on&nbsp;Berrizbeitia&nbsp;et al.&nbsp;(2020) (Belgium and Berlin annual average)&nbsp;but would be better off parametrised. | Update&nbsp;Spartacus.nml
I | top_flux_dn_diffuse_sw | Top-of-canopy downwelling diffuse longwave flux | Kdown&nbsp;-&nbsp;Top_flux_dn_direct_sw | -
SU | air_ext&nbsp;(sw) | Wavelength-independent air extinction coefficient (m-1)&nbsp;(i.e.&nbsp;the number of radiance e-foldings&nbsp;per metre). | 0.0&nbsp;(not set in&nbsp;offline Spartacus Russell_square.nc and has no default value in the code).&nbsp;It is a&nbsp;profile&nbsp;but we will treat as constant. | Update&nbsp;Spartacus.nml
SU | air_ssa&nbsp;(sw) | Shortwave single scattering albedo of air | 0.0&nbsp;(not set in&nbsp;offline Spartacus Russell_square.nc and has no default value in the code).&nbsp;It is a&nbsp;profile&nbsp;but we will treat as constant. | Update&nbsp;Spartacus.nml
SU | veg_ssa&nbsp;(sw) | Shortwave single scattering albedo of leaves | 0.13&nbsp;(just took from offline Spartacus Russell_square.nc)&nbsp;It is a&nbsp;profile&nbsp;but we will treat as constant. | Update&nbsp;Spartacus.nml
I | ground_albedo | Ground albedo | Calculated from SUEWS&nbsp;alb_next&nbsp;and weighted by&nbsp;sfr. Excludes buildings&nbsp;and&nbsp;trees. | Update&nbsp;the SUEWS_Water.txt,&nbsp;SUEWS_Veg.txt&nbsp;and&nbsp;SUEWS_NonVeg.txt&nbsp;files.
SU | roof_albedo | Roof albedo | SUEWS&nbsp;alb_next(2) | Update&nbsp;Spartacus.nml&nbsp;&nbsp;&nbsp;Check&nbsp;building albedo&nbsp;in SUEWS_NonVeg.txt is the&nbsp;same as&nbsp;roof_albedo&nbsp;first array index.
SUP | wall_albedo | Wall albedo | SUEWS&nbsp;alb_next(2) | Update&nbsp;Spartacus.nml&nbsp;&nbsp;&nbsp;Check&nbsp;building albedo&nbsp;in SUEWS_NonVeg.txt is the same as&nbsp;wall_albedo&nbsp;first array index.
SU | ground_albedo_dir_mult_fact&nbsp;&nbsp;(ground_albedo_dir=&nbsp;ground_albedo_dir_mult_fact&nbsp;*ground_albedo) | Ratio of shortwave albedo of ground to direct beam&nbsp;and diffuse beam | Not&nbsp;used&nbsp;when&nbsp;use_sw_direct_albedo&nbsp;= False | Update&nbsp;Spartacus.nml
SUP | roof_albedo_dir_mult_fact&nbsp;&nbsp;(roof_albedo_dir=&nbsp;roof_albedo_dir_mult_fact&nbsp;*roof_albedo) | Ratio of shortwave albedo of&nbsp;roof&nbsp;to direct beam&nbsp;and diffuse beam | Not&nbsp;used&nbsp;when&nbsp;use_sw_direct_albedo&nbsp;= False | Update&nbsp;Spartacus_Profiles.nml
SUP | wall_specular_frac | Fraction of wall reflection that is specular | 0.0&nbsp;as in&nbsp;the spartacus_surface_documanetation.pdf.&nbsp;Using&nbsp;values larger than 0.0 would require knowledge of the wall percentage window cover, and the specular fraction of a window. | Update&nbsp;Spartacus_Profiles.nml
SU | air_ext&nbsp;(lw) | Wavelength-independent air extinction coefficient (m-1)&nbsp;(i.e.&nbsp;the number of radiance e-foldings&nbsp;per metre) | 0.0&nbsp;(not set in&nbsp;offline Spartacus Russell_square.nc and has no default value in the code)&nbsp;It is a profile&nbsp;in&nbsp;Spartacus&nbsp;but we will treat as constant. | Update&nbsp;Spartacus.nml
SU | air_ssa&nbsp;(lw) | Longwave single scattering albedo of air | 0.0&nbsp;(not set in&nbsp;offline Spartacus Russell_square.nc and has no default value in the code).&nbsp;It is a profile&nbsp;in&nbsp;Spartacus&nbsp;but we will treat as constant. | Update&nbsp;Spartacus.nml
SU | veg_ssa&nbsp;(lw) | Longwave single scattering albedo of leaves | (just took from offline Spartacus&nbsp;Russell_square.nc)&nbsp;It is a profile&nbsp;in&nbsp;Spartacus&nbsp;but we will treat as constant. | Update&nbsp;Spartacus.nml
I | ground_emissivity | Ground emissivity | Calculated from SUEWS&nbsp;emis&nbsp;and weighted by&nbsp;sfr. Excludes buildings&nbsp;and&nbsp;trees. | -
SUP | roof_emissivity | Roof emissivity | SUEWS&nbsp;emis(2) | Update&nbsp;Spartacus_Profiles.nml&nbsp;&nbsp;&nbsp;Check&nbsp;building&nbsp;emissivty&nbsp;in SUEWS_NonVeg.txt is the same as&nbsp;roof_emissivity&nbsp;first array index.
SUP | wall_emissivity | Wall emissivity | SUEWS&nbsp;emis(2) | Update&nbsp;Spartacus_Profiles.nml&nbsp;&nbsp;&nbsp;Check&nbsp;building emissivity&nbsp;in SUEWS_NonVeg.txt is the same as&nbsp;wall_emissivity&nbsp;first array index.

**Comment by sunt05** [comment]
### with concerns


- [ ] `nspec` , SU (maybe I since lw_emission calculations are not implemented for multiple bands in Spartacus yet)  , Number of shortwave or longwave spectral intervals , 1 (should be the same as nsw and nlw) , Update `Spartacus.nml `
- [ ] `height` , SUP , Height of the layer interfaces , Nlay+1 evenly separated layers between 0 and max building height. , Update Spartacus_Profiles.nml   *** Fig to show difference between wind and temperature layers  -- which goes to RSL ***
- [ ] `veg_ext` , SUP , Wavelength independent vegetation extinction coefficient (m-1) (i.e. the number of radiance e-foldings per metre).  Should this be related to LAI? At least one parameter in Spartacus should be. Hogan et al. 2018: “If the leaves can be assumed randomly oriented then the optical depth of the layer is equal to half the leaf area index (LAI)” , 0.25 (just took from offline Spartacus Russell_square.nc) , Update Spartacus_Profiles.nml
- [ ] `veg_fsd` , SUP , Fractional standard deviation of vegetation extinction , 0.5 (just took from offline Spartacus config.nam) , Update Spartacus_Profiles.nml
- [ ] `veg_contact_fraction` , SUP , Fraction of vegetation edge in contact with building walls , 0.0 (choosing anything other than zero seems arbitrary without knowledge of the canopy geometry) , Update Spartacus_Profiles.nml
- [ ] `air_ext` (sw), SU , Wavelength-independent air extinction coefficient (m-1) (i.e. the number of radiance e-foldings per metre). , 0.0 (not set in offline Spartacus Russell_square.nc and has no default value in the code). It is a profile but we will treat as constant. , Update `Spartacus.nml `
- [ ] `air_ssa` (sw), SU , Shortwave single scattering albedo of air , 0.0 (not set in offline Spartacus Russell_square.nc and has no default value in the code). It is a profile but we will treat as constant. , Update `Spartacus.nml `
- [ ] `veg_ssa` (sw), SU , Shortwave single scattering albedo of leaves , 0.13 (just took from offline Spartacus Russell_square.nc) It is a profile but we will treat as constant. , Update `Spartacus.nml `
- [ ] `wall_specular_frac` , SUP , Fraction of wall reflection that is specular , 0.0 as in the spartacus_surface_documanetation.pdf. Using values larger than 0.0 would require knowledge of the wall percentage window cover, and the specular fraction of a window. , Update Spartacus_Profiles.nml
- [ ] `air_ext` (lw) , SU , Wavelength-independent air extinction coefficient (m-1) (i.e. the number of radiance e-foldings per metre) , 0.0 (not set in offline Spartacus Russell_square.nc and has no default value in the code) It is a profile in Spartacus but we will treat as constant. , Update `Spartacus.nml `
- [ ] `air_ssa` (lw) , SU , Longwave single scattering albedo of air , 0.0 (not set in offline Spartacus Russell_square.nc and has no default value in the code). It is a profile in Spartacus but we will treat as constant. , Update `Spartacus.nml `
- [ ] `veg_ssa` (lw) , SU , Longwave single scattering albedo of leaves , (just took from offline Spartacus Russell_square.nc) It is a profile in Spartacus but we will treat as constant. , Update `Spartacus.nml `

**Comment by sunt05** [comment]
@LewisB7 please see my comments above; for those you agree, please mark them as `done`. then we address the others.

**Comment by LewisB7** [comment]
Thanks for this Ting.

For the height array there are no concerns. Sue asked me to make a figure comparing the vertical geometry of the canopy as represented by Spartacus and RSL. I left the note in red.

I will look into veg_ext and veg_fsd. I am not yet sure how to relate them to LAI. Particularly veg_fsd since I do not know what it represents - the documentation and papers do not explain.  I think it should be easy to separate out tree types, since we can calculate the parameters for each tree type and then combine them in an area weighted average.

I have checked the other .nc files and there is no value for air_ssa. I will leave it as zero.

**Comment by rarygit** [comment]
Just browsing by: veg_fsd

Does it relate to this?
 ! Fractional standard deviation of vegetation optical depth    real(kind=jprb), allocatable :: veg_fsd(:) ! (ntotlay)
https://github.com/ecmwf/spartacus-surface/blob/eed4a4255df9d3a34fc39e14a571802bf5b3ef3c/radsurf/radsurf_canopy_properties.F90

Which has interesting, but possibly irrelevant (non-spartacus), discussions here:
The global long-term microwave Vegetation Optical Depth Climate Archive
https://essd.copernicus.org/articles/12/177/2020/ 
The use of reflection from vegetation for estimating broken-cloud optical depth
https://www.researchgate.net/publication/228413589_The_use_of_reflection_from_vegetation_for_estimating_broken-cloud_optical_depth

**Comment by rarygit** [comment]
Hi @LewisB7, that is quite a comprehensive summary, thanks! I will read through it. 
Having read it, very informative! 

As an interested bystander as to how spartacus will function within suews, I can only follow how this develops.

If it helps, Vezy 2018 made some interesting observations regarding canopy complexity and the utility of  constant extinction coefficients (https://vezy.github.io/DynACof/reference/Light_extinction_K.html) using MAESPA. In particular canopy temperature, which was my interest in simulating with MAESPA.

**Comment by rarygit** [comment]
@LewisB7 , if it helps, I used the TRY plant trait database to populate the vegetation/tree parameters in MAESPA
https://www.try-db.org/TryWeb/Home.php 
Plant traits - morphological, anatomical, biochemical, physiological or phenological features of individuals or their component organs or tissues

Remy Vezy's PhD and associated fortran code was very useful in understanding the theory and functional relationships within the MAEPA model.  https://gitlab.com/VEZY/MAESPA

I compiled a non-parallel version of his model.

Kerry Nice's PhD ran a version coupling MAESPA with TUF-3D (Krayenhoff and Voogt, 2007)) in an urban setting. 
https://users.monash.edu.au/~kerryn/files/WSC2014Poster_KerryNice.pdf
The code and examples are available online.  PhD dissertation also.
https://github.com/mothlight/VTUF-3D
The simulation output is surprisingly large (gigabytes in disk space). 
I have to check whether Kerry changed the code so that the meteorological input for MAESPA includes L_down radiation (e.g. from adjacent buildings).
